File: backend/models/comment.go
package models

import (
	"time"
)

type Comment struct {
	ID          uint      `gorm:"primarykey" json:"id"`
	Content     string    `gorm:"type:text;not null" json:"content"`
	UserID      uint      `gorm:"index" json:"user_id"`
	User        User      `gorm:"foreignKey:UserID" json:"user"`
	ArticleID   uint      `gorm:"index" json:"article_id"`
	ParentID    *uint     `gorm:"index" json:"parent_id"`
	LikeCount   int       `gorm:"default:0" json:"like_count"`
	ReplyCount  int       `gorm:"default:0" json:"reply_count"`
	Replies     []Comment `gorm:"-" json:"replies"`
	IsAnonymous bool      `gorm:"default:false" json:"is_anonymous"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CommentLike struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	UserID    uint      `gorm:"index;uniqueIndex:idx_user_comment" json:"user_id"`
	CommentID uint      `gorm:"index;uniqueIndex:idx_user_comment" json:"comment_id"`
	CreatedAt time.Time `json:"created_at"`
}

type ViewHistory struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	ArticleID uint      `gorm:"index" json:"article_id"`
	IP        string    `gorm:"index;size:45" json:"ip"`
	UserID    *uint     `gorm:"index" json:"user_id"`
	CreatedAt time.Time `json:"created_at"`
}

File: app/lib/models/comment.dart
import 'user.dart';

class Comment {
  final int id;
  final String content;
  final int userId;
  final User? user;
  final int articleId;
  final int? parentId;
  final int likeCount;
  final int replyCount;
  final List<Comment> replies;
  final String createdAt;

  Comment({
    required this.id,
    required this.content,
    required this.userId,
    this.user,
    required this.articleId,
    this.parentId,
    required this.likeCount,
    required this.replyCount,
    required this.replies,
    required this.createdAt,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'] ?? 0,
      content: json['content'] ?? '',
      userId: json['user_id'] ?? json['userId'] ?? 0,
      user: json['user'] != null ? User.fromJson(json['user'] as Map<String, dynamic>) : null,
      articleId: json['article_id'] ?? json['articleId'] ?? 0,
      parentId: json['parent_id'] != null ? (json['parent_id'] as num).toInt() : null,
      likeCount: json['like_count'] ?? json['likeCount'] ?? 0,
      replyCount: json['reply_count'] ?? json['replyCount'] ?? 0,
      replies: (json['replies'] as List<dynamic>?)
              ?.map((item) => Comment.fromJson(item as Map<String, dynamic>))
              .toList() ??
          [],
      createdAt: json['created_at'] ?? json['createdAt'] ?? '',
    );
  }
}
